### PR TITLE
Check in out

### PR DESCRIPTION
**Pull request by thoheld**
# User Story:
As a user, I can check-in to an event that I'm signed up for, so that my attendance hours can be tracked.

# Acceptance Criteria
1. Event page displays a success message saying "You've checked in!" after the user clicks the "Check-In" button.
2. The event page displays a "Check-Out" button if the user has already checked in.
3. The user's start time has been saved to dbpersonhours, and the end-time is left NULL.

## Further Notes:
Check-in and check-out are fully functional. Users can check in and out through the view event page. The user must be signed up for the event, and it must currently be the day of the event in order to be able to check-in and out

P.S. this version also includes all of Madi's changes, it might be easier to just pull this version since it can be auto-merged.

### CODE REVIEW

File: database/dbPersons.php [hunk 1]
```diff
@@ -182,6 +182,67 @@ function update_volunteer_hours($eventname, $username, $new_start_time, $new_end
 	return $result;
 }
 
+/*@@@ Thomas */
+
+/* Check-in a user by adding a new row and with start_time to dbpersonhours */
+function check_in($personID, $eventID, $start_time) {
+    $con=connect();
+	$query = "INSERT INTO dbpersonhours (personID, eventID, start_time) VALUES ( '" .$personID. "', '" .$eventID. "', '" .$start_time. "')";
```
Comment (bluelightspirit) [inline database/dbPersons.php:190]: indentation does not match rest of function

File: database/dbPersons.php [hunk 2]
```diff
@@ -182,6 +182,67 @@ function update_volunteer_hours($eventname, $username, $new_start_time, $new_end
...
+	return $result;
+}
+
+/* Check-out a user by updating their end_time */
+function check_out($personID, $eventID, $end_time) {
+    $con=connect();
+    $query = "UPDATE dbpersonhours SET end_time = '" . $end_time . "' WHERE eventID = '" .$eventID. "' AND personID = '" .$personID. "' and end_time IS NULL";
+	$result = mysqli_query($con,$query);
+	mysqli_close($con);
+	return $result;
```
Comment (bluelightspirit) [inline database/dbPersons.php]: indentation does not match rest of function - remove tab suggested

File: database/dbPersons.php [hunk 3]
```diff
@@ -182,6 +182,67 @@ function update_volunteer_hours($eventname, $username, $new_start_time, $new_end
...
+}
+
+/* Check-out a user by updating their end_time */
+function check_out($personID, $eventID, $end_time) {
+    $con=connect();
+    $query = "UPDATE dbpersonhours SET end_time = '" . $end_time . "' WHERE eventID = '" .$eventID. "' AND personID = '" .$personID. "' and end_time IS NULL";
+	$result = mysqli_query($con,$query);
+	mysqli_close($con);
```
Comment (bluelightspirit) [inline database/dbPersons.php]: indentation does not match rest of function - remove tab suggested

File: database/dbPersons.php [hunk 4]
```diff
@@ -182,6 +182,67 @@ function update_volunteer_hours($eventname, $username, $new_start_time, $new_end
...
+	return $result;
+}
+
+/* Check-out a user by updating their end_time */
+function check_out($personID, $eventID, $end_time) {
+    $con=connect();
+    $query = "UPDATE dbpersonhours SET end_time = '" . $end_time . "' WHERE eventID = '" .$eventID. "' AND personID = '" .$personID. "' and end_time IS NULL";
+	$result = mysqli_query($con,$query);
```
Comment (bluelightspirit) [inline database/dbPersons.php]: indentation does not match rest of function - remove tab suggested

File: database/dbPersons.php [hunk 5]
```diff
@@ -182,6 +182,67 @@ function update_volunteer_hours($eventname, $username, $new_start_time, $new_end
 	return $result;
 }
 
+/*@@@ Thomas */
+
+/* Check-in a user by adding a new row and with start_time to dbpersonhours */
+function check_in($personID, $eventID, $start_time) {
+    $con=connect();
```
Comment (bluelightspirit) [inline database/dbPersons.php:189]: indentation does not match rest of function - add ab suggested

File: database/dbPersons.php [hunk 6]
```diff
@@ -182,6 +182,67 @@ function update_volunteer_hours($eventname, $username, $new_start_time, $new_end
...
+	$query = "INSERT INTO dbpersonhours (personID, eventID, start_time) VALUES ( '" .$personID. "', '" .$eventID. "', '" .$start_time. "')";
+    //$query = 'UPDATE dbpersonhours SET start_time = "' . $new_start_time . '", end_time = "' . $new_end_time . ' WHERE eventID = "' . $eventid . '" AND personID = "' . $username . '"';
```
Comment (bluelightspirit) [inline database/dbPersons.php]: indentation does not match rest of function - add tab suggested

Comment (bluelightspirit) [inline database/dbPersons.php]: what does this query commented out do, and why is it commented out? suggest getting rid of this if useless

File: database/dbPersons.php (hunk 2 above)
Comment (thoheld) [reply database/dbPersons.php]: I believe this issue stems from different styles of indentation (4 spaces vs. tabs). I'll address this.

File: database/dbPersons.php (hunk 3 above)
Comment (thoheld) [reply database/dbPersons.php]: Addressed above.

File: database/dbPersons.php (hunk 4 above)
Comment (thoheld) [reply database/dbPersons.php]: Addressed above.

File: database/dbPersons.php (hunk 5 above)
Comment (thoheld) [reply database/dbPersons.php:189]: Addressed above.

File: database/dbPersons.php (hunk 6 above)
Comment (thoheld) [reply database/dbPersons.php]: Addressed above.

Comment (thoheld) [reply database/dbPersons.php]: I was using that as a reference for writing the query above. At this point it's useless so I'll remove it.

**Comment by bluelightspirit** [comment]
looks good with checkin/checkout to me

**Pull request merged**